### DISCUSSION TITLE
Postman collection export to Monika not running full collection?

### DISCUSSION THREAD

**Discussion by rgraham7503** [hyperjumptech/monika#719] [General]
I created a new postman collection and exported as json. I have ran this command "monika --postman Monika_Postman.json" 
Monika loads the collection and starts to run, but just runs the first request? The collection has a total of 6 requests. So does the postman export to Monika actually work? or is it just for a single request? The documentation on medium has me believing I can export the entire collection and it will just work?? `https://medium.com/hyperjump-tech/use-existing-postman-collections-with-monika-an-alternative-solution-to-postman-monitoring-770572eedb2c`
Exported as both postman collection 2 and 2.1 both exhibit same behavior 
Version: @hyperjumptech/monika/1.8.0 darwin-x64 node-v18.0.0

![image](https://user-images.githubusercontent.com/46932722/174668548-b42010aa-9ab3-4b5f-899e-58a18f13783d.png)

**Comment by kevin-hyperjump** [comment]
Hi @rgraham7503, I've created a new issue #730 to solve this problem. Thanks for the report!

**Comment by kevin-hyperjump** [comment]
Hi @rgraham7503, we have already fixed this problem, let's update your Monika to the latest version 🧋

**Comment by rgraham7503** [comment]
Updated to latest Monika (@hyperjumptech/monika/1.9.0 darwin-x64 node-v18.0.0)
Now I get this error....doesnt even run the first request anymore.
`/usr/local/lib/node_modules/@hyperjumptech/monika/node_modules/@oclif/core/lib/errors/index.js:29`
       ` err = new cli_2.CLIError(input, options);`
              ^

`CLIError: Your Postman file contains an invalid JSON format!
    at Object.error (/usr/local/lib/node_modules/@hyperjumptech/monika/node_modules/@oclif/core/lib/errors/index.js:29:15)
    at Monika.error (/usr/local/lib/node_modules/@hyperjumptech/monika/node_modules/@oclif/core/lib/command.js:90:23)
    at Monika.run (/usr/local/lib/node_modules/@hyperjumptech/monika/lib/index.js:191:18)
    at async Monika._run (/usr/local/lib/node_modules/@hyperjumptech/monika/node_modules/@oclif/core/lib/command.js:67:22) {
  oclif: { exit: 1 },
  code: undefined
}`

**Reply by kevin-hyperjump** [reply]
Hi @rgraham7503, can you attach your Postman collection file here?

**Reply by kevin-hyperjump** [reply]
Or, run Monika using one of the collection files I provided in the PR -> #731

**Comment by rgraham7503** [comment]
@kevin-hyperjump Here is the full postman collection file. Beautify it so its human readable. As I have shown (in the images above) this works fine as a collection in Postman

`{
	"info": {
		"_postman_id": "xxxxxx",
		"name": "Monika Monitor Collection",
		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
		"_exporter_id": "xxxxx"
	},
	"item": [
		{
			"name": "GET Basic status page",
			"event": [
				{
					"listen": "test",
					"script": {
						"exec": [
							"const jsonData = pm.response.json();",
							"pm.expect(jsonData.status.description).to.eql(pm.environment.get(\"status_description\"));"
						],
						"type": "text/javascript"
					}
				}
			],
			"request": {
				"method": "GET",
				"header": [
					{
						"key": "Content-Type",
						"value": "application/x-www-form-urlencoded",
						"type": "text"
					}
				],
				"url": {
					"raw": "https://xxxxxxxx/api/v2/status.json",
					"protocol": "https",
					"host": [
						"statuspage",
						"xxxxxxxx",
						"com"
					],
					"path": [
						"api",
						"v2",
						"status.json"
					]
				}
			},
			"response": []
		},
		{
			"name": "Get Access Token",
			"event": [
				{
					"listen": "test",
					"script": {
						"exec": [
							"pm.test(\"expect status code to be 200\", function() {",
							"    pm.response.to.have.status(200);",
							"});",
							"",
							"pm.test(\"expect response body to exist\", function() {",
							"   pm.response.to.have.body();",
							"});",
							"",
							"pm.test(\"expect response body to have access_token\", function() {",
							"   pm.response.to.have.jsonBody(\"access_token\");",
							"   let jsonData = JSON.parse(responseBody);",
							"   pm.environment.set(\"access_token\", jsonData.access_token);",
							"   pm.environment.set(\"current_client_id\", pm.environment.get(\"client_id\"));",
							"   pm.environment.set(\"current_client_secret\", pm.environment.get(\"client_secret\"));",
							"});",
							"",
							"",
							"pm.test(\"expect access_token to be set\", function () { ",
							"    pm.expect(pm.environment.get(\"access_token\")).to.exist; ",
							"});",
							"",
							"",
							""
						],
						"type": "text/javascript"
					}
				}
			],
			"request": {
				"method": "POST",
				"header": [
					{
						"key": "Content-Type",
						"value": "application/x-www-form-urlencoded"
					}
				],
				"body": {
					"mode": "urlencoded",
					"urlencoded": [
						{
							"key": "grant_type",
							"value": "client_credentials",
							"type": "text"
						},
						{
							"key": "client_id",
							"value": "{{xxxxxxxx}}",
							"type": "text"
						},
						{
							"key": "client_secret",
							"value": "{{xxxxxxxx}}",
							"type": "text"
						},
						{
							"key": "scope",
							"value": "openid",
							"type": "text",
							"disabled": true
						}
					]
				},
				"url": {
					"raw": "https://xxxxxxxx/oidc/endpoint/default/token",
					"protocol": "https",
					"host": [
						"xxxxx",
						"xxxxx",
						"xxxxxxxx",
						"com"
					],
					"path": [
						"oidc",
						"endpoint",
						"default",
						"token"
					]
				}
			},
			"response": []
		},
		{
			"name": "Create User - for monitor",
			"event": [
				{
					"listen": "test",
					"script": {
						"exec": [
							"pm.test(\"expect status code to be 201\", function() {",
							"    pm.response.to.have.status(201);",
							"});",
							"",
							"pm.test(\"expect response body to exist\", function() {",
							"   pm.response.to.have.body();",
							"});",
							"",
							"pm.test(\"expect response body to be JSON\", function() {",
							"   pm.response.to.be.json;",
							"});",
							"",
							"pm.test(\"expect response body to contain id\", function() {",
							"   pm.response.to.have.jsonBody(\"id\");",
							"   let jsonData = JSON.parse(responseBody);",
							"   pm.environment.set(\"test_userid\", jsonData.id);",
							"});",
							"",
							"pm.test(\"expect test_userid to be set\", function () { ",
							"    pm.expect(pm.environment.get(\"test_userid\")).to.exist; ",
							"});",
							"",
							""
						],
						"type": "text/javascript"
					}
				}
			],
			"request": {
				"auth": {
					"type": "bearer",
					"bearer": [
						{
							"key": "token",
							"value": "{{access_token}}",
							"type": "string"
						}
					]
				},
				"method": "POST",
				"header": [
					{
						"key": "Content-Type",
						"value": "application/scim+json"
					}
				],
				"body": {
					"mode": "raw",
					"raw": "{\r\n  \"schemas\": [\r\n    \"urn:ietf:params:scim:schemas:core:2.0:User\",\r\n        \"urn:ietf:params:scim:schemas:extension:xxxxxxxx:2.0:User\",\r\n        \"urn:ietf:params:scim:schemas:extension:enterprise:2.0:User\",\r\n        \"urn:ietf:params:scim:schemas:extension:xxxxxxxx:2.0:Notification\"\r\n  ],\r\n  \"userName\": \"xxxxxxxx\",\r\n  \"title\": \"Important person\",\r\n  \"name\": {\r\n    \"familyName\": \"tester\",\r\n    \"givenName\": \"tester\",\r\n    \"middleName\": \"test\"\r\n  },\r\n  \"urn:ietf:params:scim:schemas:extension:xxxxxxxx:2.0:Notification\": {\r\n    \"notifyType\": \"NONE\"\r\n  },\r\n  \"displayName\": \"xxxxxxxx\",\r\n  \"preferredLanguage\": \"en-US\",\r\n  \"active\": true,\r\n  \"emails\": [\r\n    {\r\n      \"value\": \"xxxxxxxx\",\r\n      \"type\": \"work\"\r\n    }\r\n  ],\r\n  \"addresses\": [\r\n    {\r\n      \"locality\": \"xxxxxxxx\",\r\n      \"country\": \"US\",\r\n      \"type\": \"work\",\r\n      \"streetAddress\": \"111 my street. mytown, mystate\",\r\n      \"postalCode\": \"xxxxxxxx\",\r\n      \"primary\": true,\r\n      \"region\": \"NY\"\r\n    }\r\n  ],\r\n  \"phoneNumbers\": [\r\n    {\r\n      \"value\": \"xxxxxxxx\",\r\n      \"type\": \"mobile\"\r\n    }\r\n  ],\r\n  \"urn:ietf:params:scim:schemas:extension:xxxxxxxx:2.0:User\": {\r\n    \"userCategory\": \"regular\",\r\n    \"twoFactorAuthentication\": false\r\n  },\r\n  \"urn:ietf:params:scim:schemas:extension:enterprise:2.0:User\": {\r\n    \"department\": \"IT Engineering\",\r\n    \"employeeNumber\": \"xxxxxxxx\"\r\n  },\r\n  \"password\": \"xxxxxxxx\"\r\n}\r\n"
				},
				"url": {
					"raw": "https://xxxxxxxx/v2.0/Users",
					"protocol": "https",
					"host": [
						"xxxxx",
						"xxxxx",
						"xxxxxxxx",
						"com"
					],
					"path": [
						"v2.0",
						"Users"
					]
				}
			},
			"response": []
		},
		{
			"name": "Authenticate user Copy",
			"event": [
				{
					"listen": "test",
					"script": {
						"exec": [
							"pm.test(\"expect status code to be 200\", function() {",
							"    pm.response.to.have.status(200);",
							"});",
							"",
							"pm.test(\"expect response body to exist\", function() {",
							"   pm.response.to.have.body();",
							"});",
							"",
							"pm.test(\"expect response body to be JSON\", function() {",
							"   pm.response.to.be.json;",
							"});",
							"",
							"",
							"",
							""
						],
						"type": "text/javascript"
					}
				}
			],
			"request": {
				"auth": {
					"type": "bearer",
					"bearer": [
						{
							"key": "token",
							"value": "{{access_token}}",
							"type": "string"
						}
					]
				},
				"method": "POST",
				"header": [
					{
						"key": "Content-Type",
						"name": "Content-Type",
						"value": "application/json",
						"type": "text"
					}
				],
				"body": {
					"mode": "raw",
					"raw": "{\r\n  \"username\": \"xxxxxxxx\",\r\n  \"password\": \"xxxxxxxx\"\r\n}"
				},
				"url": {
					"raw": "https://xxxxxxxx/v1.0/authnmethods/password/xxxxxxxx?returnJwt=true",
					"protocol": "https",
					"host": [
						"xxxxx",
						"xxxxx",
						"xxxxxxxx",
						"com"
					],
					"path": [
						"v1.0",
						"authnmethods",
						"password",
						"xxxxxxxx"
					],
					"query": [
						{
							"key": "returnJwt",
							"value": "true"
						}
					]
				}
			},
			"response": []
		},
		{
			"name": "#Get user by userName",
			"event": [
				{
					"listen": "test",
					"script": {
						"exec": [
							"pm.test(\"expect status code to be 200\", function() {",
							"    pm.response.to.have.status(200);",
							"});",
							"",
							"pm.test(\"expect response body to exist\", function() {",
							"   pm.response.to.have.body();",
							"});",
							"",
							"pm.test(\"expect response body to have totalResults\", function() {",
							"   pm.response.to.have.jsonBody(\"totalResults\"); ",
							"});",
							"",
							"pm.test(\"expect totalResults = 1\", function() {",
							"   pm.expect(pm.response.json().totalResults).to.equal(1);",
							"   let jsonData = JSON.parse(responseBody);",
							"   pm.environment.set(\"test_userid\", jsonData.Resources[0].id);",
							"});",
							"",
							"pm.test(\"expect test_userid to be set\", function () { ",
							"    pm.expect(pm.environment.get(\"test_userid\")).to.exist; ",
							"});",
							""
						],
						"type": "text/javascript"
					}
				}
			],
			"request": {
				"auth": {
					"type": "bearer",
					"bearer": [
						{
							"key": "token",
							"value": "{{access_token}}",
							"type": "string"
						}
					]
				},
				"method": "GET",
				"header": [],
				"url": {
					"raw": "https://{{hostname}}/v2.0/Users?filter=userName+co+\"xxxxxxxx\"",
					"protocol": "https",
					"host": [
						"{{hostname}}"
					],
					"path": [
						"v2.0",
						"Users"
					],
					"query": [
						{
							"key": "filter",
							"value": "userName+co+\"xxxxxxxx\""
						}
					]
				}
			},
			"response": []
		},
		{
			"name": "CLEANUP: Delete User",
			"event": [
				{
					"listen": "test",
					"script": {
						"exec": [
							"pm.test(\"expect status code to be 204\", function() {",
							"    pm.response.to.have.status(204);",
							"    pm.environment.unset('test_userid');",
							"});",
							""
						],
						"type": "text/javascript"
					}
				}
			],
			"request": {
				"auth": {
					"type": "bearer",
					"bearer": [
						{
							"key": "token",
							"value": "{{access_token}}",
							"type": "string"
						}
					]
				},
				"method": "DELETE",
				"header": [],
				"body": {
					"mode": "raw",
					"raw": ""
				},
				"url": {
					"raw": "{{scheme}}://{{hostname}}/v2.0/Users/{{test_userid}}",
					"protocol": "{{scheme}}",
					"host": [
						"{{hostname}}"
					],
					"path": [
						"v2.0",
						"Users",
						"{{test_userid}}"
					]
				}
			},
			"response": []
		}
	]
}
`

**Comment by rgraham7503** [comment]
I'm getting the same error on a simple postman get v2.0 collection
`{ "info": { "_postman_id": "xxxxx", "name": "Monika Monitor Collection", "schema": "https://schema.getpostman.com/json/collection/v2.0.0/collection.json", "_exporter_id": "xxxxxx" }, "item": [ { "name": "GET Basic status page", "request": { "method": "GET", "header": [ { "key": "Content-Type", "value": "application/x-www-form-urlencoded", "type": "text" } ], "url": "https://xxxxx/api/v2/status.json" }, "response": [] } } `

**Reply by kevin-hyperjump** [reply]
Hmm, looks like your JSON missed some brackets. I've checked using https://jsonformatter.org/

**Comment by rgraham7503** [comment]
This seems like its due to the copy/pase or me redacting information. I just validated the postman collection json file with the json validator and its valid. Is there a way I can privately send you this vs cp/paste on git? 100% the json is valid on the actual postman export.

**Reply by kevin-hyperjump** [reply]
Finally, I found and fixed the bug here: #762. Anyway, thanks for the report :)
